meson.build

```diff
@@ -546,16 +546,18 @@ configure_file(input : 'tools/toolbox.py',
 	       output : 'toolbox.py',
 	       configuration : config_ratbagctl_devel)
 # ratbagctl.devel starts a custom ratbagd and interacts with that over DBus
-configure_file(input : 'tools/ratbagctl.devel.in',
+configure_file(input : 'tools/ratbagctl.devel.py.in',
 	       output : 'ratbagctl.devel',
 	       configuration : config_ratbagctl_devel)
 # ratbagctl.test starts a custom ratbagd and interacts with that over DBus
-configure_file(input : 'tools/ratbagctl.test.in',
+configure_file(input : 'tools/ratbagctl.test.py.in',
 	       output : 'ratbagctl.test',
 	       configuration : config_ratbagctl_devel)
 
 env_test = environment()
 env_test.set('LIBRATBAG_DATA_DIR', libratbag_data_dir_devel)
+# See: https://github.com/mesonbuild/meson/issues/3306.
+env_test.set('MESON_SOURCE_ROOT', meson.source_root())
```

Comment:
Right, I even think meson gave warnings for these before :thinking:

Reply:
Ah, we use source_root() heavily, and meson was already giving a warning for it.
I've looked at this before, but I didn't fix it because `project_...` was only added in meson v0.56, while we use v0.50.
Ubuntu LTS 20.04 (alive until 2024-10 I think) is on v0.53, so I don't think we should upgrade yet.

Reply:
fwiw, most of libratbag's meson pre-dates a lot of meson's fancier features, we switched quite early. I think you could work around this doing this:
```
if meson.version().version_compare('>=0.56')
  project_source_root = meson.project_source_root()
else
  project_source_root = meson.source_root()
endif
```
and then use the variable everywhere we currently use `meson.source_root()`.

Reply:
Or since you are in the toplevel directory simply use current_source_dir.

Reply:
I ended up using a version comparison - I think it's more self-documenting
https://github.com/libratbag/libratbag/pull/1437/commits/1d24aa71d367231d2de0ac3eac59ba9eb91c09e5